mlir/test/lib/Pass/TestConvertToSPIRVPass.cpp

```diff
@@ -71,9 +66,44 @@ void populateConvertToSPIRVPatterns(const SPIRVTypeConverter &typeConverter,
 }
 
 /// A pass to perform the SPIR-V conversion.
-struct ConvertToSPIRVPass final
-    : impl::ConvertToSPIRVPassBase<ConvertToSPIRVPass> {
-  using ConvertToSPIRVPassBase::ConvertToSPIRVPassBase;
+struct TestConvertToSPIRVPass final
+    : PassWrapper<TestConvertToSPIRVPass, OperationPass<>> {
+  Option<bool> runSignatureConversion{
+      *this, "run-signature-conversion",
+      llvm::cl::desc(
+          "Run function signature conversion to convert vector types"),
+      llvm::cl::init(true)};
+  Option<bool> runVectorUnrolling{
+      *this, "run-vector-unrolling",
+      llvm::cl::desc(
+          "Run vector unrolling to convert vector types in function bodies"),
+      llvm::cl::init(true)};
+  Option<bool> convertGPUModules{
+      *this, "convert-gpu-modules",
+      llvm::cl::desc("Clone and convert GPU modules"), llvm::cl::init(false)};
+  Option<bool> nestInGPUModule{
+      *this, "nest-in-gpu-module",
+      llvm::cl::desc("Put converted SPIR-V module inside the gpu.module "
+                     "instead of alongside it."),
+      llvm::cl::init(false)};
+
+  MLIR_DEFINE_EXPLICIT_INTERNAL_INLINE_TYPE_ID(TestConvertToSPIRVPass)
+
+  StringRef getArgument() const final { return "test-convert-to-spirv"; }
+  StringRef getDescription() const final {
+    return "Conversion to SPIR-V pass only used for internal tests.";
+  }
+  void getDependentDialects(DialectRegistry &registry) const override {
+    registry.insert<spirv::SPIRVDialect>();
+    registry.insert<vector::VectorDialect>();
+  }
+
+  TestConvertToSPIRVPass() = default;
+  TestConvertToSPIRVPass(bool convertGPUModules, bool nestInGPUModule) {
+    this->convertGPUModules = std::move(convertGPUModules);
+    this->nestInGPUModule = std::move(nestInGPUModule);
```

Comment:
There's no point in calling `move` on bools. Also, prefer initialization lists in constructors.

Reply:
These can't be set with the initialization list, you have to use the assign operator as far as I can tell.

Reply:
Why is that? I'd expect this to work:
```suggestion
  TestConvertToSPIRVPass(bool convertGPUModules, bool nestInGPUModule)
    : convertGPUModules(convertGPUModules), nestInGPUModule(nestInGPUModule) {
```

Reply:
Oh, I understand -- these are initialized first and then set to a specific value. Nerve mind then.

Reply:
Maybe remove the moves only

Reply:
I tried that! But the constructor for `Option` is only for declaring the option's name and description and so on; there isn't a constructor for instead initializing it with just a value (because then it'd have no name and description, maybe?).

Reply:
Basically, it has a weird dual role of holding the value of the option and holding a description of the option, and this is the place where it trips you up.